### DISCUSSION TITLE
Distributions, versioning, and migrations

### DISCUSSION THREAD

**Discussion by DamianReeves** [finos/morphir#55] [General]
Distributions are a great and distinguishing feature in morphir. I might be missing something, but it seems that there is no version information in what is done.

Is that the case, or am I missing something?

I do feel versioning is really important for the distributions as a whole, but also for modules,  types and functions. Tagging nodes with version info will help in tracking changes over time.

**Comment by AttilaMihaly** [comment]
Yes, there is no version information. The reason is simply that we haven't decided on a package management strategy yet so I wanted to avoid getting into versions until we get there. But I think you are right, we need to start thinking about it.

For the whole IR I was following the approach of keeping the identifiers separate from the actual value. For example type and value definitions don't have a name. The name is a level higher on the module level. The motivation to create a `Distribution` on top of `Package` was the same: `Package` does not have a name in it, the name of the `Package` is in the `Distribution` type. 

`Distribution` is a weird structure though because it supposed to encapsulate multiple packages, each with a specific version but how those versions are managed would be up to the package management strategy we choose. 

We could simply add a version number but I'm not sure how that would affect package management. Do you have any specific requirements? Maybe that would help to drive our thinking in the right direction.
